File: 3078/3078.cs
public class Solution {
    public int MatchAlphanumericalPatternInMa(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MatchAlphanumericalPatternInMa(root.left), MatchAlphanumericalPatternInMa(root.right));
    }
}

File: 3078/3078.c
int matchAlphanumericalPatternInMa(struct TreeNode* root) {
    if (!root) return 0;
    int l = matchAlphanumericalPatternInMa(root->left);
    int r = matchAlphanumericalPatternInMa(root->right);
    return 1 + (l > r ? l : r);
}